### PR TITLE
Clear intermediate datasets (improved version)

### PR DESCRIPTION
**Pull request by hrjn**
Found this cool macro made by PierreP, added a few tweaks:

- option to exclude shared datasets
- a bit of refactoring
- logging
- some aesthetics

### CODE REVIEW

File: clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        project = client.get_project(self.project_key)
+        all_datasets = project.list_datasets()
+        all_recipes = project.list_recipes()
+
+        # Build deduplicated lists of input/output datasets:
+        input_datasets = []
+        output_datasets = []
+        for rcp in all_recipes:
+            # List dataset names that are inputs:
+            rcp_in = [x["ref"] for x in rcp["inputs"]["main"]["items"]]
```
Comment (du-phan) [inline clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py]: This does not work if in the flow we have either: 
- Plugin recipes
- Code recipes (python/shell/...) with no inputs.

Comment (hrjn) [reply clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py]: @du-phan the breaking case with plugin recipes happens with said recipes anywhere in the flow or just at the beginning?

**Comment by hrjn** [comment]
@du-phan update done, should work now with plugins and no-input-recipes, keep me posted :)

**Comment by du-phan** [comment]
Looks good to me! I have not tested for partitioned and shared dataset though.

**Pull request merged**